### PR TITLE
Sync subscription code from rhel 9

### PR DESCRIPTION
**Pull request by M4rtinK**
This PR syncs the RHEL 9 subscription code, which includes:
- Satellite support
- better UX for Simple Content Access (SCA)
- various subscription related fixes that were not yet upstream-ed for technical reasons

Also as entitlement support has been dropped in RHEL 10, with only SCA being supported going forward, the PR also drops code related to parsing and displaying entitlements.

While this PR is targeting RHEL 10 first, my intention is to open an upstream PR once this one is merged, to avoid any further de-sync of the subscription support codebase going forward.

### CODE REVIEW

**Comment by M4rtinK** [comment]
**How does it look like**
Can you spot the difference ?

**Before**
previous_spoke
Without Satellite support, there is no Satellite URL entry field & it is not possible to really register to Satellite, as the provisioning script will not be run.

rhel_9_entitlements
Upon registering in entitlement mode, list of attached entitlements is shown (shown on RHEL 9).

**After**

satellite_url
Satellite URL can now be entered & the backend logic will do what is necessary to properly point the installed system to the chosen Satellite instance.

registered
The system has been registered to Satellite & is registered in SCA mode (the only mode supported on RHEL 10). Note the lack of listed entitlements, as those are no longer a thing.

hub
Registration to Satellite is reflected also in the spoke statuses visible on the summary hub.

File: pyanaconda/modules/common/task/runnable.py [hunk 1]
```diff
@@ -102,16 +110,22 @@ def _task_run_callback(self):
     @async_action_nowait
     def _task_failed_callback(self):
         """Callback for a failed task."""
+        for i in self._failed_signal._methods:
+            log.debug(i)
```
Comment (rvykydal) [inline pyanaconda/modules/common/task/runnable.py:114]: The changes in this file should go in, right?
I'd maybe do it in a separate commit, or even PR.

Comment (M4rtinK) [reply pyanaconda/modules/common/task/runnable.py]: Nope, leftover debugging output, oops. :P Should be fixed now. :)

File: pyanaconda/modules/subscription/utils.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#
+# Utility functions for network module
```
Comment (rvykydal) [inline pyanaconda/modules/subscription/utils.py:2]: for subscription module

File: pyanaconda/modules/subscription/utils.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#
+# Copyright (C) 2021 Red Hat, Inc.
```
Comment (rvykydal) [inline pyanaconda/modules/subscription/utils.py:4]: 2024

File: pyanaconda/ui/lib/subscription.py [hunk 4]
```diff
@@ -145,6 +143,17 @@ def noop(*args, **kwargs):
     pass
 
 
+def dummy_error_callback(error):
```
Comment (rvykydal) [inline pyanaconda/ui/lib/subscription.py:146]: Is this used somewhere?

File: tests/unit_tests/pyanaconda_tests/modules/subscription/test_subscription_tasks.py [hunk 5]
```diff
@@ -936,33 +976,981 @@ def test_system_purpose_json_parsing(self):
...
+        with pytest.raises(SatelliteProvisioningError):
+            task.run()
+        # make sure we did try to provision the system with
+        run_script_function.assert_called_once_with(
+            provisioning_script="foo",
+            run_on_target_system=True
+        )
+
+
+class RegisterandSubscribeTestCase(unittest.TestCase):
```
Comment (rvykydal) [inline tests/unit_tests/pyanaconda_tests/modules/subscription/test_subscription_tasks.py:1121]: a typo Registerand... -> RegisterAnd...
Feel free to ignore.

**Review by rvykydal — APPROVED** [review]
Looks good to me.
I was able to find just a few harmless typos.

File: tests/unit_tests/pyanaconda_tests/modules/subscription/test_subscription_tasks.py (hunk 5 above)
Comment (M4rtinK) [reply tests/unit_tests/pyanaconda_tests/modules/subscription/test_subscription_tasks.py:1121]: Might as well fix it. :)

File: pyanaconda/ui/lib/subscription.py (hunk 4 above)
Comment (M4rtinK) [reply pyanaconda/ui/lib/subscription.py:146]: I don't think its used anymore - this was used by an older revision of the code, but is not used anymore. So removed, thanks! :)

File: pyanaconda/modules/subscription/utils.py (hunk 3 above)
Comment (M4rtinK) [reply pyanaconda/modules/subscription/utils.py:4]: Fixed. :)

File: pyanaconda/modules/subscription/utils.py (hunk 2 above)
Comment (M4rtinK) [reply pyanaconda/modules/subscription/utils.py:2]: Fixed. :)
